ui/src/message_popup/mobile_prototype/mobile_interface.jsx

```diff
@@ -116,14 +115,17 @@ export default React.createClass({
   },
 
   addMessages(messageList){
-    var messages = _.clone(this.state.messages);
-    var messageId = this.state.messageIndex;
-    for(var messageIndex = 0; messageIndex < messageList.length; messageIndex++){
-      messageId+=1;
-      const message = _.extend({key: messageId}, messageList[messageIndex]);
-      messages.push(message);
-    }
-    this.setState({messages, messageIndex: messageId});
+    const message = {key: this.state.messageIndex+1, ...messageList.shift()};
```

Comment:
In general, it's best to avoid mutating objects like `messageList` that are passed to function unless that's explicitly called out or important for memory or performance reasons.  Modifying an argument can be particularly confusing when the method schedules future changes to the argument that will happen async.  It's difficult for the caller to know that those changes will occur or when, and if it mutates the value on its own that may not be what the code within `addMessages` expects.

I think the approach here is good - to schedule messages being added and then animate them individually, since `velocity-ui` doesn't directly support staggering with different animations on each element.  My suggestion would be to model this a bit more explicitly, rather than having that sequence modeled implicitly in the `setInterval` calls.  This might look like having `addMessages` put the new messages in state and have that be it, adding a callback to the VelocityComponent that tracks which messages have finished animating and stores that in state as `animatedMessageKeys`.  In the render code, all animated messages are rendered, along with the next message that hasn't been animated yet.  Rendering would cause the animation, and when it finishes that would lead to the next one being rendered, etc.  That would make the state changes a bit more explicit and you could remove the manual scheduling code to add new messages each 300ms.

Also, in general for you'll want to make sure that any callbacks scheduled via `setInterval` or `setTimeout` are cancelled when the component is unmounted.  Otherwise they can end up being scheduled to run, the user navigates away and the component unmounts, and then the callback runs but there's no more component left :)  Using [this mixin](https://github.com/mit-teaching-systems-lab/threeflows/blob/master/ui/src/helpers/set_interval_mixin.js) can help (or something similar).